test/v2/connectors/AmendUkPropertyAnnualSubmissionConnectorSpec.scala

```diff
@@ -0,0 +1,147 @@
+
+import scala.concurrent.Future
+
+class AmendUkPropertyAnnualSubmissionConnectorSpec extends ConnectorSpec {
+
+  val nino: String = "AA123456A"
+  val businessId: String = "XAIS12345678910"
+  val taxYear: String = "2022-23"
+
+  private val ukFhlProperty = UkFhlProperty(
+    Some(UkFhlPropertyAdjustments(
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      true,
+      Some(5000.99),
+      true,
+      Some(UkPropertyAdjustmentsRentARoom(true))
+    )),
+    Some(UkFhlPropertyAllowances(
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      None
+    ))
+  )
+
+  private val ukNonFhlProperty = UkNonFhlProperty(
+    Some(UkNonFhlPropertyAdjustments(
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      true,
+      Some(UkPropertyAdjustmentsRentARoom(true))
+    )),
+    Some(UkNonFhlPropertyAllowances(
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      Some(5000.99),
+      None,
+      Some(Seq(StructuredBuildingAllowance(
+        5000.99,
+        Some(FirstYear(
+          "2020-01-01",
+          5000.99
+        )),
+        Building(
+          Some("Green Oak's"),
+          None,
+          "GF49JH"
+        )
+      ))),
+      Some(Seq(StructuredBuildingAllowance(
+        3000.50,
+        Some(FirstYear(
+          "2020-01-01",
+          3000.60
+        )),
+        Building(
+          None,
+          Some("house number"),
+          "GF49JH"
+        )
+      )))
+    ))
+  )
+
+  val body: AmendUkPropertyAnnualSubmissionRequestBody = AmendUkPropertyAnnualSubmissionRequestBody(
+    Some(ukFhlProperty),
+    Some(ukNonFhlProperty)
+  )
+
+  val request: AmendUkPropertyAnnualSubmissionRequest = AmendUkPropertyAnnualSubmissionRequest(
+    nino = Nino(nino),
+    businessId = businessId,
+    taxYear = taxYear,
+    body = body
+  )
+
+  class Test extends MockHttpClient with MockAppConfig {
+    val connector = new AmendUkPropertyAnnualSubmissionConnector(
+      http = mockHttpClient,
+      appConfig = mockAppConfig
+    )
+
+    MockAppConfig.ifsBaseUrl returns baseUrl
+    MockAppConfig.ifsToken returns "ifs-token"
+    MockAppConfig.ifsEnvironment returns "ifs-environment"
+    MockAppConfig.ifsEnvironmentHeaders returns Some(allowedIfsHeaders)
+  }
+
+  "connector" must {
+    "put a body and return a 204" in new Test {
+      val outcome = Right(ResponseWrapper(correlationId, ()))
+
+      implicit val hc: HeaderCarrier = HeaderCarrier(otherHeaders = otherHeaders ++ Seq("Content-Type" -> "application/json"))
+      val requiredIfsHeadersPut: Seq[(String, String)] = requiredIfsHeaders ++ Seq("Content-Type" -> "application/json")
+
+      MockHttpClient
+        .put(
+          url = s"$baseUrl/income-tax/business/property/uk/annual/$nino/$businessId/$taxYear",
```

Comment:
Downstream uri is
/income-tax/business/property/annual?taxableEntityId={taxableEntityId}&incomeSourceId={incomeSourceId}&taxYear={taxYear}